util/log/log.go

```diff
@@ -34,74 +43,169 @@ func FatalOnPanic() {
 	}
 }
 
+func logKV(buf *bytes.Buffer, kvs ...[]interface{}) {
+	var i int
+	var s string
+	var kv []interface{}
+	for kvi := range kvs {
+		kv = kvs[kvi]
+		l := len(kv)
+		for i = 0; i < l; i++ {
```

Comment:
Maybe instead of treating each index separately, advance the loop by 2 at a time and consider i and i+1?